opencti-platform/opencti-graphql/src/graphql/sseMiddleware.js

```diff
@@ -190,26 +184,40 @@ const createSeeMiddleware = () => {
       },
       close: () => {
         logApp.info('[STREAM] Closing SSE channel', { clientId: channel.userId });
+        if (heartbeatInterval) clearInterval(heartbeatInterval);
         channel.expirationTime = 0;
-        try {
-          res.end();
-        } catch (e) {
-          logApp.error('[STREAM] Failing to close client', { clientId: channel.userId, error: e });
+        if (!req.finished) {
+          try {
+            res.end();
+          } catch (e) {
+            logApp.error('[STREAM] Failing to close client', { clientId: channel.userId, error: e });
+          }
         }
       },
     };
+    const heartTimer = () => {
+      if (lastEventId && !req.finished) {
+        const [idTime] = lastEventId.split('-');
+        const idDate = utcDate(parseInt(idTime, 10)).toISOString();
+        channel.sendEvent(lastEventId, 'heartbeat', idDate);
+      }
+    };
+    const heartbeatInterval = setInterval(heartTimer, 5000);
```

Comment:
5000 can be replaced by STREAM_BATCH_TIME ?

Reply:
Not the same purpose, however we could introduce a constant STREAM_HEARTBEAT